tests/index.js

```diff
@@ -47,21 +46,17 @@ describe('broccoli-jscs', function() {
       var sourcePath = 'tests/fixtures/issue-found';
       chdir(sourcePath);
 
-      var tree = new jscsTree('.', {
-        logError: function() {}
-      });
+      var tree = new jscsTree('.');
 
-      builder = new broccoli.Builder(tree);
-      return builder.build().then(function() {
-        expect(tree.bypass).to.not.be.ok();
-      });
+      expect(tree.bypass).to.not.be.ok();
     });
 
     it('uses the jscsrc as configuration', function() {
       var sourcePath = 'tests/fixtures/issue-found';
-      chdir(sourcePath);
 
-      var tree = new jscsTree('.', {
+      var tree = new jscsTree(sourcePath, {
+        persist: false,
+        configPath: path.join(sourcePath, '.jscsrc'),
```

Comment:
Agreed after rereading.

The issue being addressed is that weird things happen when we are not bypassed and we use '.' as the source directory (which isn't a thing generally supported in broccoli land).  The issue is basically that we recursively traverse the new tmp/ that is added in the new root. 

I think we can test the auto detect behavior without running a build (just check the rules after constructor is done). Then we could still chdir to test, but not hit the weird recursion issue.